examples/installation/dsf_single_account_deployment/dam.tf

```diff
@@ -55,7 +55,7 @@ module "agent_gw" {
   instance_profile_name     = var.agent_gw_instance_profile_name
 
   management_server_host_for_registration = module.mx[0].private_ip
-  management_server_host_for_api_access   = module.mx[0].public_ip
+  management_server_host_for_api_access   = module.mx[0].public_ip == null || module.mx[0].public_ip == "" ? module.mx[0].private_ip : module.mx[0].public_ip
```

Comment:
'module.mx[0].public_ip == ""' - this can't happen since you have disabled this option. Right?

If so, this reduces the readability of this fie. Please remove.

I think it is best to use coalesce function. Please consider - 
https://developer.hashicorp.com/terraform/language/functions/coalesce